### PR TITLE
Migrate to Openzeppelin test Environment

### PR DESCRIPTION
**Pull request by NicholasDotSol**
Closes: #1528
Migrating to the Openzeppelin Test environment should cut downtime requirement for JS tests significantly. 1/2 minutes compared to 9/10

- [x]  2/3 unresolved tests
- [x] Circle CI
- [x] good `.sol` test alternative/ combination. I think the `.sol` tests run normally with `truffle test` after successful .js test completion, so we could probably just chain it.  

@ngrinkevich, pushed this on draft until the above are resolved.

### CODE REVIEW

**Comment by eth-r** [comment]
Great work! For the `.sol` test problem, I've found that moving the tests to stub contracts and replacing `Assert`s with `require`s results in a ~5x reduction in the time it takes to run them: #1549

**Comment by pdyraga** [comment]
```
  1) KeepRandomBeaconOperator/DkgMisbehavior
       should be able to save group members based on misbehaved data:
     Error: Returned error: VM Exception while processing transaction: revert Malformed signatures array -- Reason given: Malformed signatures array.
     at PromiEvent (node_modules/@truffle/contract/lib/promievent.js:9:30)
      at TruffleContract.submitDkgResult (node_modules/@truffle/contract/lib/execute.js:169:26)
      at Context.it (test/random_beacon_operator/TestDkgMisbehavior.js:120:28)
      at processTicksAndRejections (internal/process/task_queues.js:86:5)

  2) TokenStaking
       eligibleStake
         should report no eligible stake for passed future undelegation:
     AssertionError: expected 2000000000000000000000000 to equal 0
      at Context.it (test/token_stake/TestTokenStake.js:808:35)
      at processTicksAndRejections (internal/process/task_queues.js:86:5)
```

:(

File: solidity/test/TestBLS.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
+const blsData = require("./helpers/data.js")
+const {contract, web3} = require("@openzeppelin/test-environment")
+const {expectRevert} = require("@openzeppelin/test-helpers")
+const BLS = contract.fromArtifact('BLS');
+var assert = require('chai').assert
 
-const BLS = artifacts.require('./cryptography/BLS.sol');
-
-contract('BLS', function() {
+describe('TestBLS', function() {
```
Comment (pdyraga) [inline solidity/test/TestBLS.js:7]: This one should be `describe('BLS'` as we are describing `BLS` behavior here, not the test behavior.

File: solidity/test/TestOperatorParams.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
-const OperatorParamsStub = artifacts.require('./stubs/OperatorParamsStub.sol');
+const {contract, web3} = require("@openzeppelin/test-environment")
+var assert = require('chai').assert
+const OperatorParamsStub = contract.fromArtifact('OperatorParamsStub');
 
-contract('OperatorParamsStub', (accounts) => {
+describe('OperatorParamsStub', () => {
```
Comment (pdyraga) [inline solidity/test/TestOperatorParams.js:5]: This one should describe `OperatorParams`, the fact we need to test through the stub is an implementation detail.

File: solidity/test/altbn128/TestAltBn128.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
 const { expectRevert } = require("@openzeppelin/test-helpers")
+const { contract } = require("@openzeppelin/test-environment")
+const AltBn128Stub = contract.fromArtifact("AltBn128Stub")
 
-const AltBn128Stub = artifacts.require("./stubs/AltBn128Stub.sol")
-
-contract("AltBn128", () => {
+describe("TestAltBn128", () => {
```
Comment (pdyraga) [inline solidity/test/altbn128/TestAltBn128.js:5]: We should have `describe('AltBn128'` here as we are describing `AltBn128` behaviour.

File: solidity/test/helpers/initContracts.js [hunk 4]
```diff
@@ -36,36 +38,45 @@ async function initContracts(KeepToken, TokenStaking, KeepRandomBeaconService,
...
+
+  const reimbursements = await Reimbursements.new({from: accounts[0]});
+
+  await KeepRandomBeaconOperator.detectNetwork()
   await KeepRandomBeaconOperator.link("GroupSelection", groupSelection.address);
+  await KeepRandomBeaconOperator.detectNetwork()
   await KeepRandomBeaconOperator.link("Groups", groups.address);
+  await KeepRandomBeaconOperator.detectNetwork()
   await KeepRandomBeaconOperator.link("DKGResultVerification", dkgResultVerification.address);
+  await KeepRandomBeaconOperator.detectNetwork()
```
Comment (pdyraga) [inline solidity/test/helpers/initContracts.js:63]: Why do we need to repeat `detectNetwork` after each `.link`?

Comment (ngrinkevich) [reply solidity/test/helpers/initContracts.js:63]: oops looks like bad search and replace

**Comment by pdyraga** [comment]
`16:36` vs `06:28` for the entire `cd solidity && npm run test` step on CircleCi, nice!

**Pull request merged**